src/app-layer-dns-udp.c

```diff
@@ -269,11 +269,15 @@ static int DNSUDPResponseParse(Flow *f, void *dstate,
         }
     }
 
-    /* see if this is a "no such name" error */
-    if (ntohs(dns_header->flags) & 0x0003) {
-        SCLogDebug("no such name");
+    /* parse rcode, e.g. "noerror" or "nxdomain" */
+    uint8_t rcode = ntohs(dns_header->flags) & 0x0F;
+    if (rcode <= 10 || (rcode >= 16 && rcode <= 22)) {
```

Comment:
Where do these numbers come from? Do they relate to DNS_RCODE_*? If so, please use those.